pilot/pkg/util/informermetric/informerutil.go

```diff
@@ -0,0 +1,60 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package informermetric
+
+import (
+	"sync"
+
+	"k8s.io/client-go/tools/cache"
+
+	"istio.io/pkg/log"
+	"istio.io/pkg/monitoring"
+)
+
+var (
+	clusterLabel = monitoring.MustCreateLabel("cluster")
+
+	errorMetric = monitoring.NewSum(
+		"controller_sync_errors",
+		"Total number of errorMetric syncing controllers.",
+	)
+
+	mu       sync.RWMutex
+	handlers = map[string]cache.WatchErrorHandler{}
+)
+
+// ErrorHandlerForCluster fetches or creates an ErrorHandler that emits a metric
+// and logs when a watch error occurs. For use with SetWatchErrorHandler on SharedInformer.
+func ErrorHandlerForCluster(clusterID string) cache.WatchErrorHandler {
+	mu.RLock()
+	handler, ok := handlers[clusterID]
+	mu.RUnlock()
+	if ok {
+		return handler
+	}
+
+	mu.Lock()
+	defer mu.Unlock()
+	metric := errorMetric.With(clusterLabel.Value(clusterID))
+	if err := metric.Register(); err != nil {
```

Comment:
Do I need to register each variant with the cluster label? 
If so, can I register a "root" metric with the label and no value?

Reply:
Yeah you just need to register the metric it self, likely in an init function. No need to register every variant of time series.

Reply:
ty!